pkg/verifier/aws/aws_verifier.go

```diff
@@ -325,3 +325,133 @@ func (a *AwsVerifier) writeDebugLogs(log string) {
 	a.Output.AddDebugLogs(log)
 	a.Logger.Debug(context.TODO(), log)
 }
+
+// CreateSecurityGroup creates a security group with the specified name and cluster tag key in a specified VPC
+func (a *AwsVerifier) CreateSecurityGroup(ctx context.Context, tags map[string]string, name, vpcId string) (*ec2.CreateSecurityGroupOutput, error) {
+	input := &ec2.CreateSecurityGroupInput{
+		GroupName:   awsTools.String(name),
+		VpcId:       &vpcId,
+		Description: awsTools.String("osd-network-verifier security group"),
+	}
+	output, err := a.AwsClient.CreateSecurityGroup(ctx, input)
+	if err != nil {
+		return &ec2.CreateSecurityGroupOutput{}, err
+	}
+
+	a.Logger.Info(ctx, "Created security group with ID: %s", *output.GroupId)
+
+	// Wait up to 1 minutes for the security group to exist
+	waiter := ec2.NewSecurityGroupExistsWaiter(a.AwsClient)
+	if err := waiter.Wait(ctx, &ec2.DescribeSecurityGroupsInput{GroupIds: []string{*output.GroupId}}, 1*time.Minute); err != nil {
+		if _, err := a.AwsClient.DeleteSecurityGroup(ctx, &ec2.DeleteSecurityGroupInput{
```

Comment:
Will this work if the SG never gets created? I think it would just error saying "SG doesn't exist" and cause the egress test to fail.

Reply:
@anispate will test this by 

* On line 343 (above this comment), we can try this by adding something like `time.Sleep(1 * time.Minute)`
* Then test ONV, let it run, it’ll create a security group and then sleep for 1 minute
* During this time, have AWS ready and delete the security group in the UI
* See what error ONV returns when it wakes up from sleep and tries to describe/delete the security group. Improve the error message if needed

Reply:
printing out errors faced during the execution:
 -  network verifier error: error performing ec2:DeleteSecurityGroup: The security group 'sg-0631479082ecd80c6' does not exist
Failure!

Hi Taylor, throws this error.

Reply:
Yeah, this is what I expected but I think its a bad message to send, since the real error is the SG was unable to create.

Reply:
So, the way. i tested it is by what Michael defined above....

SG was created then i went to AWS UI and deleted the SG from there which resulted in the SG doesn't exist. 

But, initially, the SG was created.

Reply:
Right, but you were emulating the case where the SG wasn't actually created in the first place.

Reply:
This is a bit different I think, it's an edge case where AWS receives and validates a request to create security group and then fails to do so within one minute. We could prepend any errors in this block with "expected security group %s to exist, but did not find it within 1 minute" or something? Or maybe just add debug logs when we create, wait, and delete security groups?

In the more common case where AWS receives a request to create a security group and ONV is missing permissions or it's a malformed request, a relevant error would be returned about failing to create the security group

Reply:
It was taken care of as part of the latest PR.